File: home/most-wanted-letter.py
from collections import Counter
def checkio(text,all_letters=False):
	t=[]
	for e in text.lower():
		if e.islower(): t.append(e)
	c=Counter(t)
	r=[e[0] for e in sorted(c.most_common(),key=lambda e:(-e[1],e[0]))]
	return ''.join(r) if all_letters else r[0]

if __name__ == '__main__':
	assert checkio("Hello World!") == "l", "Hello test"
	assert checkio("How do you do?") == "o", "O is most wanted"
	assert checkio("One") == "e", "All letter only once."
	assert checkio("Oops!") == "o", "Don't forget about lower case."
	assert checkio("AAaooo!!!!") == "a", "Only letters."

File: home/most-wanted-letter.js
"use strict";

function mostWanted(data){
	data=data.toLowerCase();
	var t={};
	for(var e of data){
		if(e!=e.toUpperCase()){
			t[e]=t[e]||0;
			t[e]++;
		}
	}
	var r=0,sr;
	for(var k in t){
		if(t[k]>r || t[k]==r&&sr>k)r=t[k],sr=k;
	}
	return sr;
}

var assert = require('assert');

if (!global.is_checking) {
	assert.equal(mostWanted("Hello World!"), "l", "1st example");
	assert.equal(mostWanted("How do you do?"), "o", "2nd example");
	assert.equal(mostWanted("One"), "e", "3rd example");
	assert.equal(mostWanted("Oops!"), "o", "4th example");
	assert.equal(mostWanted("AAaooo!!!!"), "a", "Letters");
	console.log("Coding complete? Click 'Check' to review your tests and earn cool rewards!");
}